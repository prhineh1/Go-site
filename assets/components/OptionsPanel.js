import React from 'react';

const OptionsPanel = (props) => (
    <div className="optionsPanel">
        { Object.values(props.players.red).reduce((acc, cur) => !!cur || !!acc) && <div>Black Wins</div> }
        { Object.values(props.players.black).reduce((acc, cur) => !!cur || !!acc) && <div>Red Wins</div> }
        <button onClick={props.concede}>Concede</button>
        <div>{props.turn}'s turn</div>
    </div>
)

export default OptionsPanel;