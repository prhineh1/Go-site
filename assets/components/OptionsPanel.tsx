import React from 'react';
import PropTypes from 'prop-types';

const OptionsPanel = ({ players, concede, turn }) => (
  <div className="optionsPanel">
    { players.red.concede && <div>Black Wins</div> }
    { players.black.concede && <div>Red Wins</div> }
    <button type="button" onClick={() => concede(turn)}>Concede</button>
    <div>
      {`${turn}'s turn`}
    </div>
  </div>
);

OptionsPanel.propTypes = {
  players: PropTypes.exact({
    red: PropTypes.exact({
      concede: PropTypes.bool,
      lost: PropTypes.number,
    }),
    black: PropTypes.exact({
      concede: PropTypes.bool,
      lost: PropTypes.number,
    }),
  }).isRequired,
  concede: PropTypes.func.isRequired,
  turn: PropTypes.oneOf(['black', 'red']).isRequired,
};

export default OptionsPanel;
