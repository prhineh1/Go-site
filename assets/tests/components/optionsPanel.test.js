import React from 'react';
import OptionsPanel from '../../components/OptionsPanel';
import { shallow } from 'enzyme';
import { initState } from '../fixtures/fixtures';

let concede, wrapper, playerState;

beforeEach(() => {
    playerState = { red: initState.red, black: initState.black };
    concede = jest.fn();
    wrapper = shallow(<OptionsPanel players={playerState} concede={concede} turn='black' />);
});

test('should render OptionsPanel with initState', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render OptionsPanel with black win', () => {
    let blackWins = shallow(<OptionsPanel players={{...playerState, red: { ...initState.red, concede: true }}} concede={concede} />);
    expect(blackWins).toMatchSnapshot();
});

test('should render OptionsPanel with red win', () => {
    let redWins = shallow(<OptionsPanel players={{...playerState, black: { ...initState.black, concede: true }}} concede={concede} />);
    expect(redWins).toMatchSnapshot();
});

test('should handle concede', () => {
    wrapper.find('button').simulate('click');
    expect(concede).toHaveBeenCalled();
});
