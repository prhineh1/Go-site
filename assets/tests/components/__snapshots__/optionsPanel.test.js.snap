// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OptionsPanel with black win 1`] = `
<div
  className="optionsPanel"
>
  <div>
    Black Wins
  </div>
  <button
    onClick={[MockFunction]}
  >
    Concede
  </button>
  <div>
    's turn
  </div>
</div>
`;

exports[`should render OptionsPanel with initState 1`] = `
<div
  className="optionsPanel"
>
  <button
    onClick={[MockFunction]}
  >
    Concede
  </button>
  <div>
    black
    's turn
  </div>
</div>
`;

exports[`should render OptionsPanel with red win 1`] = `
<div
  className="optionsPanel"
>
  <div>
    Red Wins
  </div>
  <button
    onClick={[MockFunction]}
  >
    Concede
  </button>
  <div>
    's turn
  </div>
</div>
`;
